### ISSUE TITLE
Use real edge cost function

### ISSUE THREAD

**Issue by dabreegster** [nptscot/npw#29]
Summarising from https://github.com/nptscot/internal/issues/57#issuecomment-2339114096... optimal routes through the edited network will be defined only on edges, factoring in:

- edge length
- edge gradient (I've used [this](https://github.com/Urban-Analytics-Technology-Platform/od2net/blob/30471869b6efca801839efc851f7d59766d4af6f/od2net/src/network/mod.rs#L167) courtesy Rosa before)
- the infra type

@mvl22, now that the new infra types have some examples wired up, are you in a better position to provide some sample quietness weights for these?

I'm still a little unsure if the editor needs to include multiple routing profiles (fast, quiet, balanced); will revisit the notes on the directness and % of NPT routes metrics. Maybe sample quietness and speed values for the infra types will be enough, but we still need to decide how to form the final scalar edge cost. Edge length and gradient can provide a unit of time, and speed would do that too. Quietness is the more interesting thing to consider -- we could have a profile that totally bans quietness values below some value, or multiply by a scaling factor, etc.

After all of this is initially wired up, we can re-evaluate if there's time and need to do anything more detailed. Turn costs at junctions would be one idea, and counting crossings over routes (to penalise speed) is another.

**Comment by mvl22** [comment]
I'll write a clear definition for each of infrastructure types, that properly defines how each _would be_ represented in OSM, to be agreed.

We can then supply the equivalent values for each of the different route type strategies.

**Comment by mvl22** [comment]
I propose these infrastructure types would be equivalent to:

**Segregated Track (wide):**

- highway=cycleway
- cycleway=track
- foot=no
- width=5
- lanes=2
- oneway=no
- surface=asphalt
- lit=yes
- lcn=yes

**Off Road Cycleway:**

Same as 'Segregated Track (wide)' except:

- foot=yes
- width=3
- segregated=no
- No lcn tag


**Segregated Track (narrow):**

Same as 'Segregated Track (wide)' except:

- width=2
- No lanes tag


**Shared Footway:**

- highway=footway
- bicycle=yes
- width=2.5
- segregated=no
- oneway=no
- surface=asphalt

**Painted Cycle Lane:**

- highway={primary|secondary|tertiary} - NB will get different values for each variant
- cycleway=lane
- width=1.5 (or maybe 1.2?)
- lit=yes
- cycleway:separation=no

Any comments @Robinlovelace / @dabreegster?

**Comment by Robinlovelace** [comment]
Sounds fine as a starter for 10.

**Comment by Robinlovelace** [comment]
One thing that we could do: find average 'busynance' values for each category.

**Comment by dabreegster** [comment]
This all sounds reasonable to me. Just checking, our threshold for wide/narrow segregated is 2m? (https://github.com/nptscot/osmactive/blob/b08d91b310187c6b344d3682e040e47ce2519be1/R/osmactive.R#L421)

**Comment by Robinlovelace** [comment]
> This all sounds reasonable to me. Just checking, our threshold for wide/narrow segregated is 2m? (https://github.com/nptscot/osmactive/blob/b08d91b310187c6b344d3682e040e47ce2519be1/R/osmactive.R#L421)

Correct, we're going with the minimum, not absolute minimum. And it's equal to or greater than 2 (`x >= 2`) for it to be considered wide.

**Comment by Robinlovelace** [comment]
There is a function to convert the random char strings in OSM into a numeric somewhere. NA means unlikely to be wide.

**Comment by Robinlovelace** [comment]
What's the next step, starter-for-10 .json file with costs per category?

**Comment by mvl22** [comment]
> What's the next step, starter-for-10 .json file with costs per category?

Yes, at this end we will compute all the values that the CS engine would have come up with, so these infrastructure types match the existing up-front analysis, and commit a JSON file.

**Comment by mvl22** [comment]
Here is the edge costs data.

[edgecosts.json](https://github.com/user-attachments/files/17961849/edgecosts.json)

Please note that the `Mixed traffic - residential` type has a `quietness` of 40. This is an implementation-specific workaround giving a slightly artificially lower value, to deal with an engine-related shortcoming tactically (I won't go into the details). In practice it is probably sensible to increase this to 60 so it is above 50%.
